File: apps/wizarr-frontend/src/types/api/auth/MFA.ts
export type MFAList = MFA[];

export interface MFA {
    attestation: string;
    created: string;
    credential_id: string;
    id: number | string;
    name: string;
    public_key: string;
    sign_count: number;
    transports: string;
    user_id: string;
}

File: apps/wizarr-backend/wizarr_backend/app/models/database/mfa.py
from peewee import SQL, CharField, DateTimeField, IntegerField
from app.models.database.base import BaseModel


class MFA(BaseModel):
    id = IntegerField(primary_key=True)
    name = CharField()
    user_id = CharField()
    credential_id = CharField(unique=True)
    public_key = CharField(unique=True)
    sign_count = IntegerField()
    attestation = CharField(unique=True)
    transports = CharField()
    created = DateTimeField(constraints=[SQL("DEFAULT (datetime('now'))")])
